### PR TITLE
Cleanup `CesiumGltfComponent` to facilitate multi-component creation

### PR DESCRIPTION
**Pull request by j9liu**
### Summary

This PR:

- Reorganizes `loadPrimitiveGameThreadPart`  in `CesiumGltfComponent.cpp` into smaller, reusable helper functions
- Separates `CesiumGltfInstancedComponent` into its own `.h` and `.cpp` file, for consistency with other component classes.
- Does a style pass on the fields of `CesiumPrimitiveData`.

### Rationale

`CesiumGltfComponent.cpp` has become a fairly bloated file over the years. Functions like `loadPrimitiveGameThreadPart` are especially unwieldly, and tend to get overloaded as we add support for more diverse data types.

While developing support for primitive edges -- which involves a "derived" `CesiumGltfLinesComponent` created alongside a main `CesiumGltfPrimitiveComponent` -- I found I had to repeat a lot of the boilerplate code, which would blow up this function even more. This PR attempts to reorganize the code to be more digestible, or at least more reusable, in support of #1809.

## Issue number or link

N/A

## Author checklist

- [X] I have submitted a [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) (only needed once).
- [ ] I have done a full self-review of my code.
- ~[ ] I have updated `CHANGES.md` with a short summary of my change (for user-facing changes).~
- ~[ ] I have added or updated unit tests to ensure consistent code coverage as necessary.~
- ~[ ] I have updated the documentation as necessary.~

## Testing plan

Spotchecking all of the Cesium for Unreal Samples just to make sure I haven't broken anything would be great :)

## Reviewer checklist

Thank you for taking the time to review this PR. By approving a PR you are taking as much responsibility for these changes as the author.

As you review, please go through the checklist below:

- [ ] Review and run all parts of the test plan on this branch and verify it matches expectations.
    - [ ] If the issue is a bug please make sure you can reproduce the bug in the main branch and then checkout this branch to make sure it actually solved the issue.
- [ ] Review the code and make sure you do not have any remaining questions or concerns. You should understand the code change and the chosen approach. If you are not confident or have doubts about the code, please do not hesitate to ask questions.
    - [ ] Code should follow the [C++ Style Guide](https://cesium.com/learn/cesium-native/ref-doc/style-guide.html)
- [ ] Review the unit tests and make sure there are no missing tests or edge cases.
- [ ] Review documentation changes and updates to `CHANGES.md` to make sure they accurately cover the work in this PR.
- [ ] Verify that the [Contributor License Agreement](https://github.com/CesiumGS/community/tree/main/CLAs) has been submitted, if needed.

### CODE REVIEW

File: Source/CesiumRuntime/Private/CesiumPrimitive.h [hunk 1]
```diff
@@ -24,47 +25,80 @@ struct MeshPrimitive;
 } // namespace CesiumGltf
 
 /**
- * Data that is common to the Cesium mesh component classes.
+ * Data that is common between Cesium glTF component classes.
  */
 class CesiumPrimitiveData {
 public:
+  /**
```
Comment (j9liu) [inline Source/CesiumRuntime/Private/CesiumPrimitive.h:32]: Sorry this is so noisy; the gist of this is moving the `static` stuff on top and decapitalizing the other fields.

File: Source/CesiumRuntime/Private/CesiumPrimitive.h [hunk 2]
```diff
@@ -24,47 +25,80 @@ struct MeshPrimitive;
...
+   * exponent of coordinate values, not the mantissa, in order to reduce the
+   * chances of losing precision.
+   */
+  static constexpr double positionScaleFactor = 1024.0;
+
+  /**
+   * The scale matrix to apply to positions in the glTF primitive, derived from
+   * positionScaleFactor.
+   */
+  static constexpr glm::dmat4 positionScaleMatrix = glm::dmat4(
```
Comment (david-lively) [inline Source/CesiumRuntime/Private/CesiumPrimitive.h:53]: I'd rather see the `1.0 / positionScaleFactor` calculated once and then reused, but since it's `constexpr` it doesn't really matter unless someone comes along later and messes up the matrix initialization code.

**Review by david-lively — APPROVED** [review]
LGTM.
